flake.nix

```diff
@@ -15,51 +15,62 @@
       let
         pkgs = nixpkgs.legacyPackages.${system};
         poetry2nixPkg = poetry2nix.legacyPackages.${system};
+
+        controllerOverrides = final: prev:
+          let
+            addNativeBuildInputs = drvName: inputs: {
+              "${drvName}" = prev.${drvName}.overridePythonAttrs (
+                old: {
+                  nativeBuildInputs =
+                    (old.nativeBuildInputs or [ ]) ++ inputs;
+                }
+              );
+            };
+          in
+          { }
+          // addNativeBuildInputs "mpi4py" [ final.cython ]
+          // addNativeBuildInputs "rexfw" [ final.poetry-core ]
+          // addNativeBuildInputs "chainsail-helpers" [ final.poetry-core ];
+
+        controllerOpts = {
+          projectDir = ./app/controller;
+          overrides = poetry2nixPkg.overrides.withDefaults(controllerOverrides);
+          preferWheels = true;
```

Comment:
I discovered that it does seem to work to set `preferWheels` here. I assume that poetry2nix does some kind of post-processing of the precompiled `.whl` files that it downloads.

Reply:
It defaults to `false`. My understanding is that it downloads sources (and sdist in Python parlance) and builds them.

Reply:
FWIW, I imagine that it's safer to rebuild the package from scratch in a Nix environment. On the other hand, I recall you saying that you were surprised that something compiled when you used the dev shell some time back, so I thought I'd try this option. One of the reasons will be that when using traditional Python tooling (e.g. `pip`), it automatically favours installing wheels (precompiled dependencies). Using `preferWheels` with poetry2nix seems popular when I review other projects on GitHub (though I once noticed that someone disabled it specifically for macOS/darwin). I imagine that poetry2nix defaults to recompiling because of a safety factor. However, I don't know all the trade-offs. For instance, I don't know what kind of post-processing is done on the wheels to make them work with Nix, but I assume that some kind of post-processing is required to adjust the precompiled libraries to use Nix store paths.

We can switch it back to `preferWheels = false`. Whichever is fine with me until I am better informed on it 🤷. If we set up a Cachix cache, then devs can share the Nix-compiled dependencies (once pushed by CI), and so would obviate somewhat the desire to use precompiled wheels.

Reply:
Thanks for these detailed explanations! If there's precedence for that option elsewhere in projects, and it works, and it might save some build time, then let's definitely leave it activated :slightly_smiling_face: